### DISCUSSION TITLE
Azure Monitor Handle partial results

### DISCUSSION THREAD

**Discussion by msbannert** [grafana/grafana#38328] [Data Sources]
**What would you like to be added**:
Kusto Queries might deliver only partial results due to resource limitations, see [here](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/concepts/querylimits). When running such queries directly in the azure portal, one gets a notification that the results might not be complete, but if executed within grafana one does not get this information.
Note that it is not only about too much results , but also for too complex queries which require too much resources for execution, even if the result set is relatively small.
This is very problematic and makes Grafana Dashboards unreliable since you never know if the data shown is correct.

Proposal for improvement: Visualize those query issues somehow.

This image shows parts of the json response in case a kusto query results in such an error:
![image](https://user-images.githubusercontent.com/24464176/100874240-8ffdfc00-34a4-11eb-9a21-98acf55074a1.png)

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by mitulk** [comment]
@msbannert Any idea whether this is specific to Azure Monitor (i.e. if other data sources already support this) or if this is a feature request for Grafana itself?

**Reply by joshhunt** [reply]
This is mostly a feature request for the Azure Monitor data source. As long as the Azure APIs return the partial data, we can return that to the panel along with a warning that there are missing results.

We might run into some UI limitations of showing a useful-enough message though.

**Comment by msbannert** [comment]
Yes, fully agree. The Azure Monitor data source needs to handle this return value from Azure API. And how to visualize this will be a challenge.

@mitulk I do not  know of any other data sources showing a similar behaviour. To my knowledge its only the azure data source which offers this third state: result OK --  result partially OK --  failure.

**Comment by andresmgot** [comment]
I am able to reproduce this. I believe this is a bug rather than a feature request so I created https://github.com/grafana/grafana/issues/50246 to track the progress. Please use this issue for further discussion if needed!
